### PR TITLE
Exo player update

### PR DESCRIPTION
**Pull request by Mecharyry**
## Problem
We are out of date with `exo-player` 😱 

## Solution
Update `no-player` to use the latest version of `exo-player`. https://github.com/google/ExoPlayer/blob/release-v2/RELEASENOTES.md

Most notable from the releases are:
- Workaround broken AAC decoders on Galaxy S6
- Fix memory leak when seeking with repeated periods
- Fix potential IndexOutOfBoundsException when calling ExoPlayer.getDuration
- Fixed adaptive track selection logic for live playbacks

### CODE REVIEW

File: core/src/main/java/com/novoda/noplayer/internal/exoplayer/forwarder/OnCompletionForwarder.java [hunk 1]
```diff
@@ -28,6 +28,11 @@ public void onRepeatModeChanged(@Player.RepeatMode int repeatMode) {
         // TODO: should we send?
     }
 
+    @Override
+    public void onShuffleModeEnabledChanged(boolean shuffleModeEnabled) {
```
Comment (Mecharyry) [inline core/src/main/java/com/novoda/noplayer/internal/exoplayer/forwarder/OnCompletionForwarder.java]: These listeners are less than ideal but we plan to work on tidying these up as per https://github.com/novoda/no-player/issues/95

File: core/build.gradle [hunk 2]
```diff
@@ -3,8 +3,8 @@ apply plugin: 'bintray-release'
 apply plugin: 'jacoco'
 
 android {
-    compileSdkVersion 25
-    buildToolsVersion "25.0.3"
+    compileSdkVersion 27
+    buildToolsVersion "26.0.2"
```
Comment (rock3r) [inline core/build.gradle]: Why not 27.0.1 at this point, given the compile is 27?

File: demo/build.gradle [hunk 3]
```diff
@@ -1,13 +1,13 @@
 apply plugin: 'com.android.application'
 
 android {
-    compileSdkVersion 25
-    buildToolsVersion "25.0.3"
+    compileSdkVersion 27
+    buildToolsVersion "26.0.2"
```
Comment (rock3r) [inline demo/build.gradle]: Ditto here, `27.0.1`

File: gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -1,6 +1,6 @@
-#Tue Oct 25 13:38:18 BST 2016
+#Tue Dec 05 11:00:47 GMT 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-3.5-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.1-all.zip
```
Comment (rock3r) [inline gradle/wrapper/gradle-wrapper.properties]: Why not 4.3.1?

**Pull request merged**